Makefile

```diff
@@ -0,0 +1,80 @@
+.DEFAULT_GOAL := help
+
+SHELL := /bin/bash -euo pipefail
+
+REPO_ROOT := $(CURDIR)
+
+INTERACTIVE := $(shell [ -t 0 ] && echo 1)
+
+ifneq ($(wildcard $(REPO_ROOT)/.pre-commit-config.yaml),)
+	PRE_COMMIT_CONFIG_FILE ?= $(REPO_ROOT)/.pre-commit-config.yaml
+else
+	PRE_COMMIT_CONFIG_FILE ?= $(REPO_ROOT)/repo-infra/.pre-commit-config.yaml
+endif
+
+.PHONY: clean
+clean: ## remove files created during build
+	$(call print-target)
+	cd tests && rm -f coverage.*
+
+.PHONY: install-tools
+install-tools: ## go install tools
+	$(call print-target)
+	cd tools && go install -v $(shell cd tools && go list -f '{{ join .Imports " " }}' -tags=tools)
+
+.PHONY: lint
+lint: ## golangci-lint
+lint: install-tools
+	$(call print-target)
+	cd tests && golangci-lint run -c ${REPO_ROOT}/.golangci.yml --fix
+
+.PHONY: test
+test: ## go test with race detector and code coverage
+test: install-tools
+	$(call print-target)
+	cd tests && gotestsum \
+			--junitfile junit-report.xml \
+			--junitfile-testsuite-name=relative \
+			--junitfile-testcase-classname=short \
+			-- \
+			-covermode=atomic \
+			-coverprofile=coverage.out \
+			-race \
+			-short \
+			-v \
+			./...
+	cd tests && go tool cover -html=coverage.out -o coverage.html
+
+.PHONY: mod-tidy
+mod-tidy: ## go mod tidy
+	$(call print-target)
+	cd tests && go mod tidy
+	cd tools && go mod tidy
+
+.PHONY: diff
+diff: ## git diff
```

Comment:
just wondering if we need this one? and is the below `RES` line supposed to `exit 1` there? (isn't clear to me what that line is accomplishing)

Reply:
probably not! I started this Makefile from https://github.com/mesosphere/golang-repository-template which had this target, I'll take it out!

Reply:
removed in 9467b06fb7032f7c92b30592d0c1a2b192c4b681